paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -313,12 +337,26 @@ def sync_all_secrets(
             )
         )
 
+        sync_service_secrets["database-credentials"].append(
```

Comment:
@RobbySingh @nemacysts I'm going to insist that we split this cron job up now. I think I mentioned it in the last addition (boto or crypto?). At a minimum let's put this new addition into a separate cron job. It will make them easier to debug, we can set different alerting/monitoring thresholds etc. And most importantly when one of these breaks or slows down it won't then impact the others.

Reply:
++ this is something Luis mentioned when we were designing it. may need his help in how to refactor this as I'm not too sure how it should look or how this is used. I only know there's a puppet call to it on a CRON anyway: https://sourcegraph.yelpcorp.com/search?q=context:global+repo:puppet+paasta_secrets_sync&patternType=standard&sm=0

Reply:
Great, I'm sure he'll be happy to help but feel free to have a crack yourself, it should make your release workflow safer too.

High level it's going to be: duplicate this script and just keep your part and any boilerplate/argparse etc. Then if you search the name of this script in this repo you'll find refs to it that are needed in places like setup.py and some of the debian tooling to create an entrypoint in the package. Once you have that you should be able to build a deb and test the script. Then you can merge and release safely knowing that the existing stuff is unaffected. Then iterate etc. and when happy with the script you can duplicate the thing you found in puppet to create a new cron job

Reply:
we chatted internally about this and I have a PR for splitting up the cronjobs using the work that @nurdann introduced in their paasta secrets refactor a while back - so we should be fine here since this is simply adding another `--secret-type` option :)